tests/testcore/test_cluster.go

```diff
@@ -144,7 +151,9 @@ func (a *ArchiverBase) VisibilityURI() string {
 }
 
 func (f *defaultTestClusterFactory) NewCluster(t *testing.T, options *TestClusterConfig, logger log.Logger) (*TestCluster, error) {
-	return NewClusterWithPersistenceTestBaseFactory(t, options, logger, f.tbFactory)
+	newClusterMutex.Lock()
```

Comment:
is it possible to arrange things so that creating the databases/indexes happens outside of the lock, and the lock is just from NewPortProvider to cluster.Start?

Reply:
That does look like it might be hard. Another approach:

You can just delay the `pp.Close` until right before you want the server to bind:
```
// allocate ports
// set up db, can be slow
// create test cluster
lock()
pp.Close()
cluster.Start()
unlock()
```
That guards just the window that the allocated ports could be stolen

Reply:
Hm, if I see this right, the database is set up with 
```
testBase.Setup(clusterMetadataConfig)
```
which takes the result from
```
cluster.NewTestClusterMetadataConfig(...
```
which requires `ClusterInformation` which uses the random ports.